modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java

```diff
@@ -0,0 +1,760 @@
+            super.deleteIdentityProvider(PORT_OFFSET_0, PRIMARY_IS_IDP_NAME);
+
+            super.deleteServiceProvider(PORT_OFFSET_1, SECONDARY_IS_SP_NAME);
+
+            deleteAddedUsersInSecondaryIS();
+
+            client.close();
+        } catch (Exception e) {
+            log.error("Failure occured due to :" + e.getMessage(), e);
+            throw e;
+        }
+    }
+
+    @Test(groups = "wso2.is", description = "Check SAML-to-OIDC federated login")
+    public void testFederatedLogin() throws Exception {
+
+        String sessionDataKeyOfSecondaryISLogin = sendSAMLRequestToPrimaryIS();
+        Assert.assertNotNull(sessionDataKeyOfSecondaryISLogin,
+                "Unable to acquire 'sessionDataKey' value in secondary IS");
+
+        String sessionDataKeyConsentOfSecondaryIS = doAuthenticationInSecondaryIS(sessionDataKeyOfSecondaryISLogin);
+        Assert.assertNotNull(sessionDataKeyConsentOfSecondaryIS, "Invalid sessionDataKeyConsent.");
+
+        String callbackURLOfPrimaryIS = doConsentApprovalInSecondaryIS(sessionDataKeyConsentOfSecondaryIS);
+        Assert.assertNotNull(callbackURLOfPrimaryIS, "Unable to acquire authorizeCallbackURL in primary IS");
+
+        String samlResponse = getSAMLResponseFromPrimaryIS(callbackURLOfPrimaryIS);
+        Assert.assertNotNull(samlResponse, "Unable to acquire SAML response from primary IS");
+
+        String decodedSAMLResponse = new String(Base64.decode(samlResponse));
+        Assert.assertTrue(decodedSAMLResponse.contains("AuthnContextClassRef"),
+                "AuthnContextClassRef is not received.");
+
+        String homepageContent = sendSAMLResponseToWebApp(samlResponse);
+        boolean isValidLogin = validateLoginHomePageContent(homepageContent);
+        Assert.assertTrue(isValidLogin, "Invalid SAML login response received by travelocity app");
+    }
+
+    @Test(groups = "wso2.is", description = "Check SAML-to-OIDC federated logout", dependsOnMethods = {
+            "testFederatedLogin"})
+    public void testLogout() throws Exception {
+
+        sendLogoutRequestToPrimaryIS();
+
+        String samlLogoutResponseToWebapp = doLogoutConsentApprovalInSecondaryIS();
+        Assert.assertNotNull(samlLogoutResponseToWebapp,
+                "Unable to acquire SAML Logout response from travelocity app");
+
+        String decodedSAMLResponse = new String(Base64.decode(samlLogoutResponseToWebapp));
+        Assert.assertNotNull(decodedSAMLResponse);
+
+        String logoutPageContent = sendSAMLResponseToWebApp(samlLogoutResponseToWebapp);
+        boolean isValidLogout = validateLogoutPageContent(logoutPageContent);
+        Assert.assertTrue(isValidLogout, "Invalid SAML Logout response received by travelocity app");
+    }
+
+    //TODO Test case for consent denial from the federated IdP during the logout.
+//    @Test(groups = "wso2.is", description = "Check SAML-to-OIDC federated logout deny-consent", dependsOnMethods = {
+//            "testFederatedLogin"})
```

Comment:
Why this is commented?

Reply:
let's add the reason for making this as todo in the comments and link to the git issue if possible

Reply:
Addressed with https://github.com/wso2/product-is/pull/10640/commits/1229a0d9d2cb9bb365a6c0aec4236198bd667a2c.